src/main/java/uk/gov/di/ipv/service/IpvSessionService.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.di.ipv.service;
+
+import uk.gov.di.ipv.persistence.DataStore;
+import uk.gov.di.ipv.persistence.item.IpvSessionItem;
+
+import java.time.Instant;
+import java.util.UUID;
+
+public class IpvSessionService {
+
+    private final DataStore<IpvSessionItem> dataStore;
+    private final ConfigurationService configurationService;
+
+    public IpvSessionService() {
+        this.configurationService = ConfigurationService.getInstance();
+        dataStore = new DataStore<>(configurationService.getIpvSessionTableName(), IpvSessionItem.class);
+    }
+
+    public IpvSessionService(DataStore<IpvSessionItem> dataStore, ConfigurationService configurationService) {
+        this.dataStore = dataStore;
+        this.configurationService = configurationService;
+    }
+
+    public IpvSessionItem generateIpvSession() {
```

Comment:
Thus far we haven't exposed the *Item classes beyond the service layer.
At this stage it's probably best to just return a string or UUID containing the ipv session id unless the session creation date time is needed by core-front. 
In which case it would be worth creating a separate class in the domain package to return from the service. So that the *Item classes have a single responsibility of facilitation reading/writing to the database and the *Handler classes don't have knowledge of the *Item classes

Reply:
Ah yeah good shout. Not 100% sure if the full session object will be needed for core-front. Most likely not as I think it will just be using the ipvSessionId and passing it around in a header. I'll change to just return the id field.